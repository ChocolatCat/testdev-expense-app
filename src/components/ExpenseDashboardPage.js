import React, {useEffect} from 'react';
import ExpenseList from './ExpenseList';
import { useSelector, useDispatch } from 'react-redux';
import { setExpenses } from '../feature/expenseSlice';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';
import { selectExpenses, selectExpensesTotal } from '../selectors/expenses';

const ExpenseDashboardPage = () => {
    const dispatch = useDispatch();
    //We make sure the read is only made once
    useEffect(() => {
        dispatch(setExpenses());
    }, [dispatch]);
    //We get the rendered expenses
    const visibleExpenses = useSelector(state => selectExpenses(state.expenses, state.filters));
    //We get them to sum them up
    const expensesCount = visibleExpenses.length;
    const expensesTotal = selectExpensesTotal(visibleExpenses);
    return(
        <div>
            <ExpenseListFilters />
            <ExpensesSummary expensesCount={expensesCount} expensesTotal={expensesTotal}/>
            <ExpenseList />
        </div>
    );
};

export default ExpenseDashboardPage;