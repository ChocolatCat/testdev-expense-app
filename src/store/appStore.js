import {configureStore} from '@reduxjs/toolkit';
import expenseReducer from '../feature/expenseSlice';
import filtersReducer from '../feature/filtersSlice';

export default () => {
    const store = configureStore({
        reducer:{
            expenses: expenseReducer,
            filters: filtersReducer
        }
    });
    return store;
}