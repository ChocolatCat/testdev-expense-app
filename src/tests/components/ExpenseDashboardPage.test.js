/** @jest-environment jsdom */

import React from 'react';
import {Provider} from 'react-redux';
import { render, screen } from '@testing-library/react';
import appStore from '../../store/appStore';
import ExpenseDashboardPage from '../../components/ExpenseDashboardPage';

const store = appStore();

//Mocks nanoid module because it's imported and used
jest.mock("nanoid", () => {
  return { nanoid: () => "1234" };
});

it('Should render ExpenseDashboardPage', () => {
    const {asFragment} = render(
        <Provider store={store}>
            <ExpenseDashboardPage />
        </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});