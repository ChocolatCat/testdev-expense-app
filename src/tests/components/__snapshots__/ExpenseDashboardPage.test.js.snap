// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseDashboardPage 1`] = `
<DocumentFragment>
  <div>
    <div>
      <input
        placeholder="Filter by Text"
        type="text"
        value=""
      />
      <select
        name="Sort By"
      >
        <option
          value="date"
        >
          Date
        </option>
        <option
          value="amount"
        >
          Amount
        </option>
      </select>
      <div
        class="react-datepicker-wrapper"
      >
        <div
          class="react-datepicker__input-container"
        >
          <input
            class=""
            type="text"
            value="01/08/2022"
          />
        </div>
      </div>
      <div
        class="react-datepicker-wrapper"
      >
        <div
          class="react-datepicker__input-container"
        >
          <input
            class=""
            type="text"
            value="31/08/2022"
          />
        </div>
      </div>
    </div>
    <div>
      <p>
        No Expenses Added!
      </p>
    </div>
  </div>
</DocumentFragment>
`;
